runtime/core/freeable_buffer.h

```diff
@@ -105,8 +175,16 @@ class FreeableBuffer final {
   FreeableBuffer& operator=(const FreeableBuffer& rhs) = delete;
 
   FreeFn free_fn_;
+  FreeUInt64Fn free_uint64_fn_;
```

Comment:
it looks like at most one free function can be set, and it needs to correspond to the contents of data_, right? if so, I would reorganize things like so:

```
struct PointerData {
  FreeFn free_fn_;
  const void* data_;
};

struct Int64Data {
  FreeUInt64Fn free_fn_;
  uint64_t data_;
};

std::variant<PointerData, Int64Data> data_;
```

this way we only increase the size of FreeableBuffer by 8 bytes instead of 16.